release/ray_release/job_manager/anyscale_job_manager.py

```diff
@@ -259,6 +261,49 @@ def run_and_wait(
         )
         return self._wait_job(timeout)
 
+    def _get_ray_error_logs(self) -> Optional[str]:
+        """
+        Obtain any ray logs that contain keywords that indicate a crash, such as
+        ERROR or Traceback
+        """
+        tmpdir = tempfile.mktemp()
+        try:
+            subprocess.check_output(
+                f"anyscale logs cluster --id {self.cluster_manager.cluster_id} "
+                f"--head-only --download --download-dir {tmpdir}",
+                shell=True,
+            )
+        except Exception as e:
+            logger.log(f"Failed to download logs from anyscale {e}")
+            return None
+        # Ignored some ray files that do not crash ray despite having exceptions
+        ignored_ray_files = [
+            "monitor.log",
+            "event_AUTOSCALER.log",
+            "event_JOBS.log",
+        ]
+        # Logs that can indicate there are crashes
+        error_log_patterns = [
+            "ERROR",
+            "Traceback (most recent call last)",
+        ]
```

Comment:
this pattern matching actually feels really useful. shall we extract this exception matching function into a standalone function, so that it is easier to reuse?

also, are there any unit tests for this?

Reply:
Good point!